src/service/history-impl.js

```diff
@@ -143,15 +143,20 @@ export class History {
   }
 
   /**
-   * Requests navigation one step back. This request is only satisifed
-   * when the history has at least one step to go back in the context
-   * of this document.
+   * Requests navigation one step back. This first attempts to go back within
+   * the context of this document. If there is no history left, then attempt
+   * the window's history.back.
+   *
+   * @param {boolean=} canPerformWindowBack
    * @return {!Promise}
    */
-  goBack() {
+  goBack(canPerformWindowBack) {
     return this.enque_(() => {
       if (this.stackIndex_ <= 0) {
         // Nothing left to pop.
+        if (canPerformWindowBack) {
+          this.ampdoc_.win.history.back();
```

Comment:
Hm, I'm not sure this is the right thing to do for `HistoryBindingVirtual`. We probably should still delegate to the viewer to avoid cross-iframe history mutation bugs -- does it work today if we remove this outer conditional?